lib/concurrent-ruby/concurrent/set.rb

```diff
@@ -23,9 +23,21 @@ module Concurrent
   # @!macro internal_implementation_note
   SetImplementation = case
                       when Concurrent.on_cruby?
-                        # Because MRI never runs code in parallel, the existing
-                        # non-thread-safe structures should usually work fine.
-                        ::Set
+                        # The CRuby implementation of Set is written in Ruby itself and is
+                        # not thread safe for certain methods.
+                        require 'monitor'
+                        require 'concurrent/thread_safe/util/data_structures'
+
+                        class CRubySet < ::Set
+                          def initialize(*args)
+                            self.send(:_mon_initialize)
```

Comment:
Without this -- certain Sets are being created not through _initialize_
I would see:
```
      when initializing with a block argument
concurrent-ruby/concurrent/thread_safe/util/data_structures.rb:43: warning: instance variable @_monitor not initialized
```

I'm not certain how the Set is being initialized without going through:
```ruby
          def self.new(*args)
            obj = super(*args)
            obj.send(:_mon_initialize)
            obj
          end
```

TBH i'm